### PR TITLE
Adopt to refactored scheduling API (lib/uksched)

### PR DESCRIPTION
**Pull request by skuenzer**
Adopts the lwIP glue code to work with the latest scheduling API (`staging-mulspreview`)

### CODE REVIEW

File: uknetdev.c [hunk 1]
```diff
@@ -384,11 +384,11 @@ static void uknetdev_updown(struct netif *nf)
...
 			/* Create a thread */
-			lwip_data->sched = uk_sched_get_default();
+			lwip_data->sched = uk_sched_current();
 			UK_ASSERT(lwip_data->sched);
 			lwip_data->poll_thread =
-				uk_sched_thread_create(lwip_data->sched, NULL,
-						       NULL, _poll_netif, nf);
+				uk_sched_thread_create(lwip_data->sched,
+						       _poll_netif, nf, NULL);
 #else /* CONFIG_HAVE_SCHED */
```
Comment (Sairajkodilkar) [inline uknetdev.c:392]: Hi @skuenzer the PR looks good to me, I have tested this PR with virtio and https server and is running without any error.
I just have one question. Why `uk_sched_thread_create()` creates a thread without any stack ?

File: threads.c [hunk 2]
```diff
@@ -46,12 +46,21 @@
...
+
+	t = uk_sched_thread_create_fn1(s,
+				       (uk_thread_fn1_t) thread, arg,
+				       (size_t) stacksize,
+				       false,
+				       false,
+				       name,
+				       NULL, NULL);
+	if (!t) {
+		uk_pr_err("Failed to create lwIP thread '%s'\n");
```
Comment (razvand) [inline threads.c]:
```suggestion
		uk_pr_err("Failed to create lwIP thread '%s'\n", name);
```

I guess that `name` is the missing argument, right?

**Review by razvand — CHANGES_REQUESTED** [review]
Thanks, @skuenzer . I added a small comment.

File: threads.c (hunk 2 above)
Comment (skuenzer) [reply threads.c]: Good point, yes. I'll fix it.

File: uknetdev.c (hunk 1 above)
Comment (skuenzer) [reply uknetdev.c:392]: Yes, this is true but `0x0` means that a stack is still allocated, just a default size is used instead.

`lib/uksched/include/uk/sched.h`:154:
```
 * @param stack_len
 *   Size of the thread stack. If set to 0, a default stack size is used
 *   for the stack allocation.
```

**Comment by skuenzer** [comment]
I just sent an update with the fix of the error message.

**Comment by eduardvintila** [comment]
Hi, @skuenzer. This looks great, no comments at all from me.

Reviewed-by: Eduard Vintilă <eduard.vintila47@gmail.com>

**Review by eduardvintila — APPROVED** [review]
Hi, @skuenzer. This looks great, no comments at all from me.

Reviewed-by: Eduard Vintilă <eduard.vintila47@gmail.com>

**Review by razvand — APPROVED** [review]
Thanks, @skuenzer 

Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request closed without merging**